crates/sui-types/src/digests.rs

```diff
@@ -596,6 +608,22 @@ impl TryFrom<&[u8]> for TransactionDigest {
     }
 }
 
+impl TryFrom<Vec<u8>> for TransactionDigest {
+    type Error = crate::error::SuiError;
+
+    fn try_from(bytes: Vec<u8>) -> Result<Self, crate::error::SuiError> {
+        let actual = bytes.len();
+        let arr: [u8; 32] =
+            bytes
+                .try_into()
+                .map_err(|_| crate::error::SuiError::InvalidDigestLength {
+                    expected: 32,
+                    actual,
+                })?;
+        Ok(Self::new(arr))
```

Comment:
any reason this one doesn't forward to Digest's impl?

Reply:
none whatsoever! fixed